hazelcast/internal/asyncio_proxy/base.py

```diff
@@ -0,0 +1,297 @@
+import abc
+import asyncio
+import typing
+import uuid
+
+from hazelcast.core import MemberInfo
+from hazelcast.types import KeyType, ValueType, ItemType, MessageType, BlockingProxyType
+from hazelcast.internal.asyncio_invocation import Invocation
+from hazelcast.partition import string_partition_strategy
+from hazelcast.util import get_attr_name
+
+MAX_SIZE = float("inf")
+
+
+def _no_op_response_handler(_):
+    return None
+
+
+class Proxy(typing.Generic[BlockingProxyType], abc.ABC):
+    """Provides basic functionality for Hazelcast Proxies."""
+
+    def __init__(self, service_name: str, name: str, context):
+        self.service_name = service_name
+        self.name = name
+        self._context = context
+        self._invocation_service = context.invocation_service
+        self._partition_service = context.partition_service
+        serialization_service = context.serialization_service
+        self._to_object = serialization_service.to_object
+        self._to_data = serialization_service.to_data
+        listener_service = context.listener_service
+        self._register_listener = listener_service.register_listener
+        self._deregister_listener = listener_service.deregister_listener
+        self._is_smart = context.config.smart_routing
+        self._send_schema_and_retry = context.compact_schema_service.send_schema_and_retry
+
+    async def destroy(self) -> bool:
+        """Destroys this proxy.
+
+        Returns:
+            ``True`` if this proxy is destroyed successfully, ``False``
+            otherwise.
+        """
+        self._on_destroy()
+        return await self._context.proxy_manager.destroy_proxy(self.service_name, self.name)
+
+    def _on_destroy(self):
+        pass
+
+    def __repr__(self) -> str:
+        return '%s(name="%s")' % (type(self).__name__, self.name)
+
+    def _invoke(self, request, response_handler=_no_op_response_handler) -> asyncio.Future:
```

Comment:
why not put `async def` but return future? i thought you would not need to use futures when you have asyncio.